File: plugins/src/main/java/plugin/buttons/PrayerButton.java
package plugin.buttons;

import com.astraeus.game.event.SubscribesTo;
import com.astraeus.game.event.impl.ButtonActionEvent;
import com.astraeus.game.world.entity.mob.combat.CombatPrayer;
import com.astraeus.game.world.entity.mob.player.Player;

@SubscribesTo(ButtonActionEvent.class)
public final class PrayerButton extends ButtonClick {

	@Override
	protected void execute(Player player, ButtonActionEvent event) {
		player.getCombatPrayer().clickButton(event.getButton());
	}

	@Override
	public boolean test(ButtonActionEvent event) {
		return CombatPrayer.isPrayerButton(event.getButton());
	}

}

File: plugins/plugin/buttons/PrayerButton.kt
package plugin.buttons

import astraeus.game.event.SubscribesTo
import astraeus.game.event.impl.ButtonActionEvent
import astraeus.game.model.entity.mob.player.Player
import astraeus.game.model.entity.mob.player.CombatPrayer

@SubscribesTo(ButtonActionEvent::class)
class PrayerButton : ButtonClick() {

    override fun execute(player: Player, event: ButtonActionEvent) {
        player.prayer.clickButton(event.button)
    }

    override fun test(event: ButtonActionEvent): Boolean {
        return CombatPrayer.isPrayerButton(event.button)
    }

}